page_objects/examples/weather_shopper_mobile_app/cart_objects.py

```diff
@@ -0,0 +1,80 @@
+"""
+This class models the objects of the cart in Weathershopper application.
+"""
+import conf.locators_conf as locators
+from utils.Wrapit import Wrapit
+import re
+
+class CartObjects:
+    """
+    Page Objects for the cart in Weather Shopper application
+    """
+
+    @Wrapit._exceptionHandler
+    @Wrapit._screenshot
+    def change_quantity(self, item_to_change, quantity=2):
+        "This method is to change the quantity of an item in the cart"
+        result_flag = self.set_text(locators.edit_quantity.format(item_to_change), quantity)
+        self.conditional_write(result_flag,
+            positive=f"Successfully changed quantity of {item_to_change} to {quantity}",
+            negative=f"Failed to change quantity of {item_to_change} to {quantity}",
+            level='debug')
+        return result_flag
+
+    @Wrapit._exceptionHandler
+    @Wrapit._screenshot
+    def checkout(self):
+        "This method is to go to the Checkout page"
+        result_flag = self.click_element(locators.checkout_button)
+        self.conditional_write(result_flag,
+            positive="Successfully clicked on checkout button",
+            negative="Failed to click on checkout button",
+            level='debug')
+        self.switch_page("weathershopper payment page")
+        return result_flag
+
+    @Wrapit._exceptionHandler
+    @Wrapit._screenshot
+    def delete_from_cart(self, item_to_delete):
+        "This method is to delete an item from the cart"
+        result_flag = self.click_element(locators.checkbox.format(item_to_delete))
+        result_flag &= self.click_element(locators.delete_from_cart_button)
+        self.conditional_write(result_flag,
+            positive=f"Successfully deleted {item_to_delete} from cart",
+            negative="Failed to delete item from cart",
+            level='debug')
+        return result_flag
+
+    @Wrapit._exceptionHandler
+    @Wrapit._screenshot
+    def get_cart_total(self):
+        "This method is to get the total price in the cart."
+        cart_total = self.get_text(locators.total_amount)
+        # Extracting the numeric part using regular expression
+        match = re.search(r'\d+\.\d+', cart_total.decode())
+        total_amount = float(match.group()) if match else None
+
+        if total_amount is None:
+            self.write("Total amount is None", level='debug')
+        else:
+            self.write(f"Total amount is {total_amount}", level='debug')
+        return total_amount
+
+    @Wrapit._exceptionHandler
+    @Wrapit._screenshot
+    def refresh_total_amount(self):
+        "This method is to refresh the total amount in the cart."
+        result_flag = self.click_element(locators.refresh_button)
+        self.conditional_write(result_flag,
+            positive="Successfully clicked on refresh button",
+            negative="Failed to click on refresh button",
+            level='debug')
+        return result_flag
+
+    @Wrapit._exceptionHandler
```

Comment:
verify_total can be part of the page instead of page objects

Reply:
Moved it to page